iiasa/ixmp
# Use ‘genno’ for reporting

This PR adjusts `ixmp` to use the new package `genno` for reporting.

The main changes are:
- `ixmp.Reporter` subclasses `genno.Computer`. This class now mainly adds the method `.from_scenario()` and the association with the extra computations defined by ixmp:
- `ixmp.reporting.computations` only contains a few operations specific to `ixmp`, including:
  - `update_scenario()`, moved upstream from `message_data`.

As a result:
- `ixmp` and its test suite are slimmed significantly, by about **20%**.  See the Codecov comment, below.
- Common reporting code, now consolidated in `genno`, is easier to maintain, no longer requiring PRs across 2–3 repos.
- We have better separation of concerns.

## How to review

- Note that CI checks all pass.
- Try the code in combination with:
  - ~khaeru/genno#28~ merged; genno >= 1.0.0
  - iiasa/message_ix#441
  - iiasa/message_data#182
- View the revised documentation and ensure it is sufficient on a first read.

## PR checklist

- [x] Expand this checklist/template to include: “CI checks all pass”.
- [x] Add or expand tests.
  - [x] Adapt test of `update_scenario()`.
  - [x] Boost coverage to pass Codecov checks.
